crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -230,13 +253,22 @@ impl ProtocolFees {
         quote: domain::Quote,
         partner_fees: Vec<Policy>,
     ) -> domain::Order {
-        let protocol_fees = self
-            .fee_policies
+        // Use new fee policies if the order creation date is after their effective
```

Comment:
I don't think this is the desired logic. As of the target date, all orders that are settled should be settled according to the new fee policies (as these are what decides the rewards that solvers will receive for these trades0.

In particular, a limit order that has been placed in the past should also carry the new fee policy going forward.

I'm not sure what the best way to achieve this is, but don't we build the fee policies from scratch on every auction? In this case I'd apply the logic like in the orderbook - as of a certain timestamp we start adding them unconditionally on when the order was placed.